front_end/src/components/history/HistoryOverview.js

```diff
@@ -1,93 +1,434 @@
 import React, { Component } from "react";
+import { withStyles, withTheme } from "@material-ui/core/styles";
 import { connect } from "react-redux";
 import PropTypes from "prop-types";
 import Spinner from "../common/Spinner";
-import { getCurrentProfile } from "../../actions/profileActions";
-import { Link } from "react-router-dom";
+import { getCurrentProfile, getHistory } from "../../actions/profileActions";
+import Grid from "@material-ui/core/Grid";
+import Chip from "@material-ui/core/Chip";
+import DoneIcon from "@material-ui/icons/Done";
+import SwapHorizIcon from "@material-ui/icons/SwapHoriz";
+import CancelIcon from "@material-ui/icons/Cancel";
+import ExitIcon from "@material-ui/icons/ExitToApp";
+
+import Button from "@material-ui/core/Button";
+import ExpansionPanel from "@material-ui/core/ExpansionPanel";
+import ExpansionPanelSummary from "@material-ui/core/ExpansionPanelSummary";
+import ExpansionPanelDetails from "@material-ui/core/ExpansionPanelDetails";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+
+import Table from "@material-ui/core/Table";
+import TableCell from "@material-ui/core/TableCell";
+import TableRow from "@material-ui/core/TableRow";
+
+import "./history.css";
+
+const styles = theme => ({
+  table: {
+    maxWidth: 330,
+    minWidth: 300,
+    maxHeight: 50
+  },
+  paddingDense: {
+    paddingTop: 0,
+    paddingBottom: 0
+  },
+  tableCell: {
+    maxHeight: 50
+  },
+  tableNoBorder: {
+    maxHeight: 10,
+
+    border: 0
+  },
+  chipChange: {
+    marginLeft: 3,
+    backgroundColor: "#FFA500"
+  },
+  chipCancel: {
+    backgroundColor: "#FF4500"
+  },
+  chipCheckin: {
+    backgroundColor: "#3ba711"
+  },
+  dateChangedFrom: {
+    textDecoration: "line-through"
+  },
+  dateChangedTo: {
+    color: "#FFA500"
+  },
+  backgroundStyle: {
+    background: "linear-gradient(45deg, #ffffff 30%, #cfe6fe 90%)"
+  }
+});
+
 class HistoryOverview extends Component {
   constructor() {
```

Comment:
We can remove constructor if there are no states and binding methods !

Reply:
done